### DISCUSSION TITLE
Debian vs CentOS

### DISCUSSION THREAD

**Discussion by PizzaProgram** [SoftEtherVPN/SoftEtherVPN#1785] [General]
According to the "[7.3 Install on Linux and Initial Configurations](https://www.softether.org/4-docs/1-manual/7._Installing_SoftEther_VPN_Server/7.3_Install_on_Linux_and_Initial_Configurations)" page the only 3 recommended Linux OS are:

- Red Hat Enterprise Linux
- Fedora
- CentOS

Is that really still the case? Or just the documentation is outdated?
If it IS possible to install it to debian _(which I know a little more than, while I never touched those other 3)_ is there any documentation how to do it?

Currently it's Debian 11 installed, and I can only reinstall the VPS with these options:
![kép](https://user-images.githubusercontent.com/26026089/221888021-059b9118-8fcb-4036-8177-cb5642374c4a.png)

... interesting thing is, that when I've ordered the VPS there was a CentOS 8.4.2105 available, but I think that still the:
- OLD "CentOS Linux 8" (EOL 2021) and 
- not the NEW "CentOS Stream 8" (EOL 2024)

![kép](https://user-images.githubusercontent.com/26026089/221890652-cf72b736-37e3-466a-8b57-dbbe095a641e.png)

**Comment by PizzaProgram** [comment]
I've just found this:
[How to Install SoftEther on Debian 9, Ubuntu 16+!](https://community.fulltimehosting.net/t/how-to-install-softether-on-debian-9-ubuntu-16/42)

I'll try it ...

**Comment by PizzaProgram** [comment]
You are right! Sorry.
I've just thought we could enhance the documentation...

Please convert this to a Discussion. (I have no rights to do it.)

**Comment by metalefty** [comment]
I think either is fine. However, please keep in mind that the document is for version 4.x also known as [SoftEtherVPN_Stable](https://github.com/SoftEtherVPN/SoftEtherVPN_Stable). It's simple and obvious but please keep in mind again, the document for 4.x is for 4.x but not for 5.x. Some are helpful also with 5.x but some are not.

Regarding 5.x, it requires libsodium which is not available on CentOS standard repository. We need to enable EPEL In addition to standard repositories. Debian is a little bit more recommended if you don't want to enable EPEL.

**Comment by metalefty** [comment]
> Is that really still the case? Or just the documentation is outdated?

I think it is not outdated. It is still up to date for the latest version of 4.x. Just it is for 4.x.

**Comment by PizzaProgram** [comment]
A quick report:

- Managed to install it to Debian 11 in ca 5 hours of trial/fail.  
- The success lied on this very well hidden sub-page: [BUILD UNIX](https://github.com/SoftEtherVPN/SoftEtherVPN/blob/master/src/BUILD_UNIX.md)  
- I've found this link at an old topic about Ubuntu install. (After searching ca 560 other topics and issues here + vpnusers.com + the entire web)
- Later I've realised it is "actually there" on the main page too, but somebody must be a genius to find that, after scrolling down 12 pages of pure text 

 ... (I've red ca. 5x the main page = `readme.md`, and 50x the first part, but still didn't realise that "installing to Linux" = "[Build from Source code](https://github.com/SoftEtherVPN/SoftEtherVPN#build-from-source-code)"

# Conclusion:
It would be great to insert +1 line at the beginning under the line "windows":
- Linux/Unix (Debian, CentOS, Fedora, Red Hat, Ubuntu, etc...)
- also [This Page](https://www.softether.org/4-docs/1-manual/7._Installing_SoftEther_VPN_Server/7.3_Install_on_Linux_and_Initial_Configurations) should have +1 paragraph (7.3.0) to the top, or at least to the bottom (7.3.10), to tell about "latest unstable DEV version nightly build install" ... also mentioning keywords of "Ubuntu / Debian / etc ... "

This 2 tiny links would have had spare me 4.5 hours of my life and a very bad "install experience".

# Present:
The server seems to running fine on Debian 11 now, I will start adding clients today.

# Concerns:
- I haven't found any topic about "How to secure a UNIX server / set up firewall, Proxy, etc" to have a SAFE VPN server.
- Need a monitoring system to check the CPU usage (currently I have 2 Gold Xeon CPU 2.9GHz)
- Haven't found here any info about the "choosable" encryption list. 
Is it top-2-down the strongest to weakest? 
- I'm worried a bit about CPU usage, with different encryption methods. It would be great to have a "Benchmark" button where somebody could test the server before he chooses ... to see how many connections could it theoretically handle. Similar way to [this...](https://veracrypt.fr/en/Home_VeraCrypt_Creating_Volume.png), but maybe that's an other topic...

**Reply by shipitsin-ilia** [reply]
what do think if 12 pages are collapsed into smaller sections ? like that https://gist.github.com/pierrejoubert73/902cc94d79424356a8d20be2b382e1ab

as for monitoring, there are good tools like Datadog or New Relic, they provide good overall observability (however, SE metrics may be added using some "integration" which does not yet exists)

**Reply by PizzaProgram** [reply]
Looks cool!
But I think that 1-1 lines I've suggested are enough.

PS: As for extending the [BUILD UNIX](https://github.com/SoftEtherVPN/SoftEtherVPN/blob/master/src/BUILD_UNIX.md) page:
- Created a [pull req.](https://github.com/SoftEtherVPN/SoftEtherVPN/pull/1790), but not excepted yet.

**Reply by shipitsin-ilia** [reply]
I cannot evaluate that improvement. It would be good if someone not familiar with SE VPN will try to install SE VPN on Debian 11 using improved doc, Your "first hand" experience is very valuable. You tried to follow doc in straightforward way and came  with feedback.

for me your PR  looks good

**Comment by PizzaProgram** [comment]
PS:
I've installed everything under root. 
Could not find/create any "ready to use" startup script for Debian. If the server restarts, I'll need to start vpnserver manually.
(I don't think this would happen many times, but who knows...)

installing under root was a little different, because the mentioned commands did not work (`sudo ...` )
instead:
```
apt-get install build-essential cmake gcc g++ make 
apt-get install git
apt-get install pkg-config
apt-get install libncurses-dev, libssl-dev libsodium-dev zlib1g-dev

git clone https://github.com/SoftEtherVPN/SoftEtherVPN.git
cd SoftEtherVPN
./configure
make .
make
make install

```
I think this whole list should be copied to the [BUILD UNIX](https://github.com/SoftEtherVPN/SoftEtherVPN/blob/master/src/BUILD_UNIX.md) page under a new chapter: Debian 11 (2023)

(`make -C build` did not work.) 
NOT found pkgconf -> pkg-config OK
libncurses5-dev is out of repo [See...](https://stackoverflow.com/questions/68964718/how-to-apt-get-install-libncurses5-dev)

**Reply by shipitsin-ilia** [reply]
pkg-config is tricky to document, I seen that myself. However, CMake provides clear evidence that it expects pkg-config
